Userland/tee.cpp

```diff
@@ -0,0 +1,96 @@
+#include <AK/Vector.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <signal.h>
+#include <stdio.h>
+#include <unistd.h>
+
+static Vector<int> collect_fds(int argc, char** argv, int start, bool aflag, bool* err)
+{
+    int oflag;
+    mode_t mode;
+    if (aflag) {
+        oflag = O_APPEND;
+        mode = 0;
+    } else {
+        oflag = O_CREAT | O_WRONLY | O_TRUNC;
+        mode = S_IROTH | S_IWOTH | S_IRGRP | S_IWGRP | S_IRUSR | S_IWUSR;
+    }
+
+    Vector<int> fds;
+    for (int i = start; i < argc; ++i) {
+        int fd = open(argv[i], oflag, mode);
+        if (fd < 0) {
+            perror("failed to open file for writing");
+            *err = true;
+        }
+        fds.append(fd);
```

Comment:
Perhaps
```suggestion
        else {
            fds.append(fd);
        }
```

Reply:
Indeed. Any `write()` with an invalid fd argument will fail with `EINVAL` so there's no point in trying.